### PR TITLE
migrate relevant articles to Vue3

### PR DESCRIPTION
**Pull request by tahmeedshah**
### General description
_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs
_Link to the related design prototypes (if applicable)_

### Testing instructions
_Provide minimal instructions on how to test this PR._

* Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
* [ ]  Branch name / PR includes the related Jira ticket Id.
* [ ]  Tests to check core implementation / bug fix added.
* [ ]  All checks in Continuous Integration workflow pass.
* [ ]  Feature functionally tested by reviewer(s).
* [ ]  Code reviewed by reviewer(s).
* [ ]  Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: app/components/kb/relevant-articles.vue [hunk 1]
```diff
@@ -2,93 +2,79 @@
...
+    import { ref, onMounted } from "vue";
+    import ArticlesApi from "./article-api";
+    import { getUrl } from '../../services/composables/articles.js';
+    import { lstring } from "./filters";
+    import { useRealm } from '../../services/composables/realm.js';
+    import { useI18n } from 'vue-i18n';
+    import messages from '../../app-text/components/kb.json';
+    const { t } = useI18n({ messages });
+    const realm = useRealm();
+    const { tag, type, sort } = defineProps({
```
Comment (blaisefonseca) [inline app/components/kb/relevant-articles.vue:31]: destructing the props will loose the reactivity, use 
```suggestion
    const props = defineProps({
```

File: app/components/kb/relevant-articles.vue [hunk 2]
```diff
@@ -2,93 +2,79 @@
...
+
+    const articleUrl = (article) => {
+        return getUrl(lstring(article.title), article._id, tag);
+    };
+
+    const shuffleArray = (array) => {
+        return array.map((value) => ({ value, sort: Math.random() * 100 }))
+                .sort((a, b) => a.sort - b.sort)
+                .map(({ value }) => value);
+    };
```
Comment (blaisefonseca) [inline app/components/kb/relevant-articles.vue:78]: move it to services

File: app/components/kb/relevant-articles.vue (hunk 1 above)
Comment (tahmeedshah) [reply app/components/kb/relevant-articles.vue]: updated as suggested.

File: app/components/kb/relevant-articles.vue (hunk 2 above)
Comment (tahmeedshah) [reply app/components/kb/relevant-articles.vue]: updated as suggested.

File: app/components/kb/relevant-articles.vue [hunk 3]
```diff
@@ -2,93 +2,74 @@
...
-        ag.push({"$match":{"$and":[{"adminTags": { $all : [this.$realm.is('BCH') ? 'bch' : 'abs' ]}}]}});
-        ag.push({"$match":{"$and":[{"adminTags":encodeURIComponent(this.tag)}]}});
+    const articles = ref([]);
+    const loading = ref(true);
+    const articlesApi = new ArticlesApi();
+
+    onMounted(async () => {
+    let ag = [];
+    ag.push({"$match":{"$and":[{"adminTags": { $all : [realm.is('BCH') ? 'bch' : 'abs' ]}}]}});
+        ag.push({"$match":{"$and":[{"adminTags":encodeURIComponent(props.tag)}]}});
```
Comment (sbilodeau) [inline app/components/kb/relevant-articles.vue:49]: i know it comes from the old file but why `props.tag` are encoded?

File: app/components/kb/relevant-articles.vue [hunk 4]
```diff
@@ -2,93 +2,74 @@
...
+  
+<script setup>
+    import { ref, onMounted } from "vue";
+    import ArticlesApi from "./article-api";
+    import { getUrl, shuffleArray } from '../../services/composables/articles.js';
+    import { lstring } from "./filters";
+    import { useRealm } from '../../services/composables/realm.js';
+    import { useI18n } from 'vue-i18n';
+    import messages from '../../app-text/components/kb.json';
+    const { t } = useI18n({ messages });
```
Comment (sbilodeau) [inline app/components/kb/relevant-articles.vue:29]:
```suggestion
    const { t, locale } = useI18n({ messages });
```

File: app/components/kb/relevant-articles.vue [hunk 5]
```diff
@@ -2,93 +2,74 @@
...
-                <hr></hr>
+            {{ t("relevantArticles") }}
+                <hr/>
             </h4>
         </slot>
-        <div class="loading" v-if="loading"><i class="fa fa-cog fa-spin fa-lg" ></i> {{ $t("loading") }}...</div>
+        <div class="loading" v-if="loading"><i class="fa fa-cog fa-spin fa-lg" ></i> {{ t("loading") }}...</div>
         <ul>
             <li v-for="article in articles" class="mb-1">
                 <a class="link-dark fs-6" :href="`${articleUrl(article)}`">{{lstring(article.title, $locale)}}</a>
```
Comment (sbilodeau) [inline app/components/kb/relevant-articles.vue:12]: I think `$locale` is legacy and should come from the ```const { locale } = useI18n()```
```suggestion
                <a class="link-dark fs-6" :href="`${articleUrl(article)}`">{{lstring(article.title, locale)}}</a>
```

**Review by sbilodeau — CHANGES_REQUESTED** [review]

File: app/components/kb/relevant-articles.vue (hunk 4 above)
Comment (tahmeedshah) [reply app/components/kb/relevant-articles.vue]: done.

File: app/components/kb/relevant-articles.vue (hunk 3 above)
Comment (tahmeedshah) [reply app/components/kb/relevant-articles.vue]: yes, we can remove it.

File: app/components/kb/relevant-articles.vue (hunk 5 above)
Comment (tahmeedshah) [reply app/components/kb/relevant-articles.vue]: done.

**Review by blaisefonseca — APPROVED** [review]

**Review by sbilodeau — APPROVED** [review]

**Pull request merged**